library/src/main/java/com/owncloud/android/lib/resources/e2ee/StoreMetadataV2RemoteOperation.kt

```diff
@@ -25,6 +27,18 @@ class StoreMetadataV2RemoteOperation(
     private val token: String,
     private val signature: String
 ) : RemoteOperation<String>() {
+    private var sessionTimeOut: SessionTimeOut = defaultSessionTimeOut
+
+    constructor(
```

Comment:
Can't we use default constructor values in this case? That way we could consolidate both constructors into one:

```kotlin
class StoreMetadataV2RemoteOperation @JvmOverloads constructor(
    private val remoteId: String,
    private val encryptedMetadataJson: String,
    private val token: String,
    private val signature: String,
    private val sessionTimeOut: SessionTimeOut = defaultSessionTimeOut
) : RemoteOperation<String>() {
    // ...
}
```

We should be able to use default values in most Kotlin classes for this pull request.

Reply:
These classes called from java files, and you have to pass sessionTimeOut parameter, thus default value cannot be used from java files. Client code shouldn't affect from these changes.

Reply:
While I fully agree, the [`@JvmOverloads`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.jvm/-jvm-overloads/) annotation used in the example instructs the compiler to generate all necessary constructors, including ones without default values.

Reply:
https://github.com/nextcloud/android-library/pull/1597/commits/635e112725fe1f195039891abcedd33fb7d2a8e2 Thanks